### PR TITLE
Add a11y role to list and spacing to list item

### LINKED ISSUE #2766: Make psammead-bulleted-list more accessible by updating element roles
**Is your feature request related to a problem? Please describe.**
There's a bug on VoiceOver that does not announce the list when `list-style-type` is set to none.

**Describe the solution you'd like**
Add `role='list'` and `role='list item'` attributes to `ul` and `li` elements respectively.

**Testing notes**
[Tester to complete]

Dev insight: Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by pharingee**
Resolves #2766

**Overall change:** 
Add `role='list'` to `BulletedList` and create `BulletedListItem` with `role='listitem'`. And add spacing to list item.

**Code changes:**
- Add `role` to `BulletedList`
- Create `BulletedListItem`
- Space out `BulletedListItem`
- Update tests
- Update stories

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by ryanmccombe — APPROVED** [review]
Looks good 👍 just needs version bump

**Review by karinathomasbbc — APPROVED** [review]
Looks good to me

**Review by jroebu14 — COMMENTED** [review]
which issue does this resolve? would be good to learn about this requirement

**Comment by greenc05** [comment]
Please could we add a note in the readme under the a11y notes about the roles? Such as:

We have added role list and role listitem to the corresponding items due to a VoiceOver bug to reinstate the list semantics

File: packages/components/psammead-bulleted-list/src/index.jsx [hunk 1]
```diff
@@ -22,6 +27,10 @@ const BulletedList = styled.ul`
   }
 `;
 
+export const BulletedListItem = styled.li.attrs(() => ({ role: 'listitem' }))`
```
Comment (jroebu14) [inline packages/components/psammead-bulleted-list/src/index.jsx:30]: Cool I didn't know that was thing you could do

**Review by jroebu14 — APPROVED** [review]

**Pull request merged**